### PR TITLE
asyncapi v0.11.2 (new formula)

### PR DESCRIPTION
**Pull request by derberg**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----
formula generated with https://github.com/zmwangx/homebrew-npm-noob, only tests added, inspired by https://github.com/Homebrew/homebrew-core/blob/master/Formula/angular-cli.rb

### CODE REVIEW

**Comment by derberg** [comment]
I haven't been able to test it locally because of error `RuntimeError: npm failed to pack`. I see that for the same reason tests fail here on CI.

I tried to find some answers online but haven't found any answers 😞

**Comment by derberg** [comment]
I wonder if it is because in `bin` dir there is just a `run` file and should be `asyncapi` file 🤔

**Comment by derberg** [comment]
I just checked another example, apollo-cli formula and they also use oclify to build CLI, and they also have `run` as name of the run file, so the same as we have -> https://github.com/apollographql/apollo-tooling/tree/master/packages/apollo/bin so all should be good on that side.

any ideas? should I open a separate issue?

**Comment by derberg** [comment]
```
/usr/local/Homebrew/Library/Homebrew/brew.rb (Formulary::FromPathLoader): loading /Users/wookiee/sources/homebrew-core/Formula/asyncapi.rb
Error: Failed to load cask: asyncapi.rb
Cask 'asyncapi' is unreadable: wrong constant name #<Class:0x00007fbf6396ff58>
Warning: Treating asyncapi.rb as a formula.
/usr/local/Homebrew/Library/Homebrew/brew.rb (Formulary::FormulaLoader): loading /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/node.rb
/usr/local/Homebrew/Library/Homebrew/brew.rb (Formulary::FormulaLoader): loading /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/pkg-config.rb
/usr/local/Homebrew/Library/Homebrew/brew.rb (Formulary::FormulaLoader): loading /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/python@3.9.rb
/usr/local/Homebrew/Library/Homebrew/brew.rb (Formulary::FormulaLoader): loading /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/gdbm.rb
/usr/local/Homebrew/Library/Homebrew/brew.rb (Formulary::FormulaLoader): loading /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/mpdecimal.rb
/usr/local/Homebrew/Library/Homebrew/brew.rb (Formulary::FormulaLoader): loading /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/openssl@1.1.rb
/usr/local/Homebrew/Library/Homebrew/brew.rb (Formulary::FormulaLoader): loading /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/ca-certificates.rb
/usr/local/Homebrew/Library/Homebrew/brew.rb (Formulary::FormulaLoader): loading /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/readline.rb
/usr/local/Homebrew/Library/Homebrew/brew.rb (Formulary::FormulaLoader): loading /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/sqlite.rb
/usr/local/Homebrew/Library/Homebrew/brew.rb (Formulary::FormulaLoader): loading /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/xz.rb
/usr/local/Homebrew/Library/Homebrew/brew.rb (Formulary::FormulaLoader): loading /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/brotli.rb
/usr/local/Homebrew/Library/Homebrew/brew.rb (Formulary::FormulaLoader): loading /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/cmake.rb
/usr/local/Homebrew/Library/Homebrew/brew.rb (Formulary::FormulaLoader): loading /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/c-ares.rb
/usr/local/Homebrew/Library/Homebrew/brew.rb (Formulary::FormulaLoader): loading /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/icu4c.rb
/usr/local/Homebrew/Library/Homebrew/brew.rb (Formulary::FormulaLoader): loading /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/libnghttp2.rb
/usr/local/Homebrew/Library/Homebrew/brew.rb (Formulary::FormulaLoader): loading /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/libuv.rb
/usr/local/Homebrew/Library/Homebrew/brew.rb (Formulary::FormulaLoader): loading /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/autoconf.rb
/usr/local/Homebrew/Library/Homebrew/brew.rb (Formulary::FormulaLoader): loading /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/m4.rb
/usr/local/Homebrew/Library/Homebrew/brew.rb (Formulary::FormulaLoader): loading /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/automake.rb
/usr/local/Homebrew/Library/Homebrew/brew.rb (Formulary::FormulaLoader): loading /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/libtool.rb
/usr/local/Homebrew/Library/Homebrew/brew.rb (Formulary::FormulaLoader): loading /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/sphinx-doc.rb
/usr/local/Homebrew/Library/Homebrew/brew.rb (Formulary::FormulaLoader): loading /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/python@3.10.rb
==> Downloading https://registry.npmjs.org/@asyncapi/cli/-/cli-0.11.2.tgz
/usr/local/Homebrew/Library/Homebrew/shims/shared/curl --disable --cookie /dev/null --globoff --show-error --user-agent Homebrew/3.3.9-69-g435e22d\ \(Macintosh\;\ Intel\ Mac\ OS\ X\ 11.6\)\ curl/7.64.1 --header Accept-Language:\ en --retry 3 --location --silent --head --request GET https://registry.npmjs.org/@asyncapi/cli/-/cli-0.11.2.tgz
Already downloaded: /Users/wookiee/Library/Caches/Homebrew/downloads/077d50a7dddd68e480889da4e75aa519fd0f95832325667752b99e5e61107732--cli-0.11.2.tgz
==> Verifying checksum for '077d50a7dddd68e480889da4e75aa519fd0f95832325667752b99e5e61107732--cli-0.11.2.tgz'
/usr/local/Homebrew/Library/Homebrew/build.rb (Formulary::FormulaLoader): loading /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/node.rb
/usr/local/Homebrew/Library/Homebrew/build.rb (Formulary::FormulaLoader): loading /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/brotli.rb
/usr/local/Homebrew/Library/Homebrew/build.rb (Formulary::FormulaLoader): loading /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/c-ares.rb
/usr/local/Homebrew/Library/Homebrew/build.rb (Formulary::FormulaLoader): loading /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/icu4c.rb
/usr/local/Homebrew/Library/Homebrew/build.rb (Formulary::FormulaLoader): loading /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/libnghttp2.rb
/usr/local/Homebrew/Library/Homebrew/build.rb (Formulary::FormulaLoader): loading /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/libuv.rb
/usr/local/Homebrew/Library/Homebrew/build.rb (Formulary::FormulaLoader): loading /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/openssl@1.1.rb
/usr/local/Homebrew/Library/Homebrew/build.rb (Formulary::FormulaLoader): loading /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/ca-certificates.rb
/usr/local/Homebrew/Library/Homebrew/build.rb (Formulary::FormulaLoader): loading /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/pkg-config.rb
/usr/local/Homebrew/Library/Homebrew/build.rb (Formulary::FormulaLoader): loading /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/python@3.9.rb
/usr/local/Homebrew/Library/Homebrew/build.rb (Formulary::FormulaLoader): loading /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/gdbm.rb
/usr/local/Homebrew/Library/Homebrew/build.rb (Formulary::FormulaLoader): loading /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/mpdecimal.rb
/usr/local/Homebrew/Library/Homebrew/build.rb (Formulary::FormulaLoader): loading /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/readline.rb
/usr/local/Homebrew/Library/Homebrew/build.rb (Formulary::FormulaLoader): loading /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/sqlite.rb
/usr/local/Homebrew/Library/Homebrew/build.rb (Formulary::FormulaLoader): loading /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/xz.rb
/usr/local/Homebrew/Library/Homebrew/build.rb (Formulary::FormulaLoader): loading /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/cmake.rb
/usr/local/Homebrew/Library/Homebrew/build.rb (Formulary::FormulaLoader): loading /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/autoconf.rb
/usr/local/Homebrew/Library/Homebrew/build.rb (Formulary::FormulaLoader): loading /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/m4.rb
/usr/local/Homebrew/Library/Homebrew/build.rb (Formulary::FormulaLoader): loading /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/automake.rb
/usr/local/Homebrew/Library/Homebrew/build.rb (Formulary::FormulaLoader): loading /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/libtool.rb
/usr/local/Homebrew/Library/Homebrew/build.rb (Formulary::FormulaLoader): loading /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/sphinx-doc.rb
/usr/local/Homebrew/Library/Homebrew/build.rb (Formulary::FormulaLoader): loading /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/python@3.10.rb
tar --extract --no-same-owner --file /Users/wookiee/Library/Caches/Homebrew/downloads/077d50a7dddd68e480889da4e75aa519fd0f95832325667752b99e5e61107732--cli-0.11.2.tgz --directory /private/tmp/d20211228-35985-1nlrqbh
cp -pR /private/tmp/d20211228-35985-1nlrqbh/package/. /private/tmp/asyncapi-20211228-35985-bhajeb/package
chmod -Rf +w /private/tmp/d20211228-35985-1nlrqbh
/usr/local/Homebrew/Library/Homebrew/ignorable.rb:29:in `block in raise'
RuntimeError: npm failed to pack /private/tmp/asyncapi-20211228-35985-bhajeb/package
1. raise
2. ignore
3. backtrace
4. irb
5. shell
Choose an action: 1
==> Temporary files retained at:
/private/tmp/asyncapi-20211228-35985-bhajeb
Error: An exception occurred within a child process:
  RuntimeError: npm failed to pack /private/tmp/asyncapi-20211228-35985-bhajeb/package
/usr/local/Homebrew/Library/Homebrew/ignorable.rb:29:in `block in raise'
/usr/local/Homebrew/Library/Homebrew/ignorable.rb:28:in `callcc'
/usr/local/Homebrew/Library/Homebrew/ignorable.rb:28:in `raise'
/usr/local/Homebrew/Library/Homebrew/language/node.rb:34:in `pack_for_installation'
/usr/local/Homebrew/Library/Homebrew/language/node.rb:59:in `std_npm_install_args'
/Users/wookiee/sources/homebrew-core/Formula/asyncapi.rb:17:in `install'
/usr/local/Homebrew/Library/Homebrew/debrew.rb:17:in `block in install'
/usr/local/Homebrew/Library/Homebrew/debrew.rb:90:in `debrew'
/usr/local/Homebrew/Library/Homebrew/debrew.rb:17:in `install'
/usr/local/Homebrew/Library/Homebrew/build.rb:172:in `block (3 levels) in install'
/usr/local/Homebrew/Library/Homebrew/utils.rb:588:in `with_env'
/usr/local/Homebrew/Library/Homebrew/build.rb:134:in `block (2 levels) in install'
/usr/local/Homebrew/Library/Homebrew/formula.rb:1297:in `block in brew'
/usr/local/Homebrew/Library/Homebrew/formula.rb:2473:in `block (2 levels) in stage'
/usr/local/Homebrew/Library/Homebrew/utils.rb:588:in `with_env'
/usr/local/Homebrew/Library/Homebrew/formula.rb:2472:in `block in stage'
/usr/local/Homebrew/Library/Homebrew/resource.rb:126:in `block (2 levels) in unpack'
/usr/local/Homebrew/Library/Homebrew/download_strategy.rb:115:in `chdir'
/usr/local/Homebrew/Library/Homebrew/download_strategy.rb:115:in `chdir'
/usr/local/Homebrew/Library/Homebrew/download_strategy.rb:102:in `stage'
/usr/local/Homebrew/Library/Homebrew/resource.rb:122:in `block in unpack'
/usr/local/Homebrew/Library/Homebrew/mktemp.rb:63:in `block in run'
/usr/local/Homebrew/Library/Homebrew/mktemp.rb:63:in `chdir'
/usr/local/Homebrew/Library/Homebrew/mktemp.rb:63:in `run'
/usr/local/Homebrew/Library/Homebrew/resource.rb:208:in `mktemp'
/usr/local/Homebrew/Library/Homebrew/resource.rb:121:in `unpack'
/usr/local/Homebrew/Library/Homebrew/resource.rb:96:in `stage'
/usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.8/lib/ruby/2.6.0/forwardable.rb:230:in `stage'
/usr/local/Homebrew/Library/Homebrew/formula.rb:2452:in `stage'
/usr/local/Homebrew/Library/Homebrew/formula.rb:1290:in `brew'
/usr/local/Homebrew/Library/Homebrew/build.rb:129:in `block in install'
/usr/local/Homebrew/Library/Homebrew/utils.rb:588:in `with_env'
/usr/local/Homebrew/Library/Homebrew/build.rb:124:in `install'
/usr/local/Homebrew/Library/Homebrew/build.rb:224:in `<main>'
```

File: Formula/asyncapi.rb [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+class Asyncapi < Formula
+  desc "All in one CLI for all AsyncAPI tools"
+  homepage "https://github.com/asyncapi/cli"
+  url "https://registry.npmjs.org/@asyncapi/cli/-/cli-0.11.2.tgz"
+  sha256 "75e4d0e73f0ea8693ccfa684c3fd5ca26cab4f4a634c9064b90520093df882bc"
+  license "Apache-2.0"
+
+  livecheck do
+    url :stable
+  end
```
Comment (carlocab) [inline Formula/asyncapi.rb:12]:
```suggestion
```
This isn't needed.

**Comment by carlocab** [comment]
@chrmoritz, any chance you could look at the `npm failed to pack` error?

**Comment by chrmoritz** [comment]
That's an issue with language node. The error comes from `postpack` trying to delete an non existing file at that point.

It can be worked around by disabling the `postpack` script and later deleting that file manually, like so

```ruby
  def install
    inreplace "package.json", "postpack", "postpack_disabled"
    system "npm", "install", *Language::Node.std_npm_install_args(libexec)
    bin.install_symlink Dir["#{libexec}/bin/*"]
    rm libexec/"lib/node_modules/@asyncapi/cli/oclif.manifest.json"
  end
```

**Comment by derberg** [comment]
@carlocab @chrmoritz thanks a lot folks, I applied the workaround, locally installation works and tests pass 💪🏼 

Thanks a bunch!

tbh I still do not get why this script is fired after all, like, when I install locally with verbose flag, these is nothing about postpack script being triggered. It is triggered by `npm` only when you call `npm pack` or `npm publish`, but definitely not `npm install`. Anyway, most important is that it works ❤️

File: Formula/asyncapi.rb [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+  sha256 "75e4d0e73f0ea8693ccfa684c3fd5ca26cab4f4a634c9064b90520093df882bc"
+  license "Apache-2.0"
+
+  depends_on "node"
+
+  def install
+    inreplace "package.json", "postpack", "postpack_disabled"
+    system "npm", "install", *Language::Node.std_npm_install_args(libexec)
+    bin.install_symlink Dir["#{libexec}/bin/*"]
+    rm libexec/"lib/node_modules/@asyncapi/cli/oclif.manifest.json"
```
Comment (carlocab) [inline Formula/asyncapi.rb:16]: Can we add some comments explaining why we're doing the `inreplace` and `rm` here?

**Comment by chrmoritz** [comment]
The reason is, that homebrew doesn't do a normal `npm instal formula -g --prefix=libexec`, because that would mean we throw away our downloaded source of the formula and let npm download everything. Currently we do some hacks to make npm reuse the Homebrew downloaded sources of the formula, which unfortunately has a few edge cases like this one, see: https://github.com/Homebrew/brew/blob/master/Library/Homebrew/language/node.rb

**Comment by derberg** [comment]
It still puzzles me why it is needed 😄 

Take apollo CLI and [their formula](https://github.com/Homebrew/homebrew-core/blob/master/Formula/apollo-cli.rb). They use oclif, also have manifest file but do not need a workaround

In their [`package.json`](https://github.com/apollographql/apollo-tooling/blob/master/packages/apollo/package.json#L26). They do `rm -f` of manifest file, so basically making sure no failure when it is not there, but we use [`rimraf`](https://www.npmjs.com/package/rimraf) that behaves exactly the same 🤷🏼

**Comment by chrmoritz** [comment]
> In their package.json. They do rm -f of manifest file, so basically making sure no failure when it is not there, but we use rimraf that behaves exactly the same 🤷🏼

They use `rm -f` and it doesn't error on a non existing file, unlike `rimraf` you are using? Not saying that this is the better behaviour, it just doesn't trigger the issue here.

**Comment by derberg** [comment]
We use `rimraf` so basically anyone can run a script on any operating system (including windows) and it will work, `rm` is just not there on windows.

but locally when I test, `rimraf` behaves exactly the same as `rm -f`, actually `rm -rf`. No errors are triggered in the console. Super strange

**Comment by derberg** [comment]
still CI fails 🤔

**Comment by derberg** [comment]
Nice! now it is all clear 😆 thanks a bunch, and 🍻 for responsiveness 🙇🏼 

Yes, CLI is just a repo, it is for tooling related to https://github.com/asyncapi/spec (almost 1.9k ⭐ ) and one of the tools that it integrates is https://github.com/asyncapi/generator (409 ⭐ ). We are just not fans of monorepo there.

**Comment by derberg** [comment]
rimraf issues solved 🥳 
only confusion about ⭐ is there

what would be the next steps?

**Comment by carlocab** [comment]
> we can ask the AsyncAPI community to add ⭐ to meet the requirement, but does it make sense if we know we have a big user base already?

It would be nice if you could do this. We regularly reject new formulae for the notability requirement, and I'd really rather not give the contributors whose PRs we reject the excuse of "but you accepted x, and they don't meet the notability requirement". (These arguments are rarely successful, but that doesn't make dealing with them any better.)

**Comment by derberg** [comment]
@carlocab Hey, so this is what we got

Screenshot 2022-01-20 at 08 56 19

Is it enough for a proof? I know you expect 30 watches but tbh I do not want to push the community to do it, even I limit my `watching` on repos and rely on totally different tools to be up to date (automation with slack).

These are the numbers from our [main repo](https://github.com/asyncapi/spec)
Screenshot 2022-01-20 at 08 59 29

And also from [previous CLI for code generation](https://github.com/asyncapi/generator) 
Screenshot 2022-01-20 at 08 59 52

I hope we won't suffer just because we do not like monorepos 😄

**Review by carlocab — APPROVED** [review]
Apologies for the delay here, @derberg. I missed your last message -- I have way too many GitHub notifications.

No need for additional "watches" -- the notability criteria is actually "any of" and not "all of", so the stars are enough.

Thanks for your contribution!

**Pull request closed without merging**